doc/release-notes/6467-optimize-permission-lookups-for-a-user.md

```diff
@@ -0,0 +1,9 @@
+The following API have been added:
+
+/api/users/{identifier}/allowedcollections/{permission}
+
+This API lists the dataverses/collections that the user has access to via the permission passed.
```

Comment:
What about cases where the user is granted access at runtime based on their membership in Shibboleth groups or IP groups?

We have this related comment in the code:

```
/**
 * We don't expect this to support Shibboleth groups because even though
 * a Shibboleth user can have an API token the transient
 * shibIdentityProvider String on AuthenticatedUser is only set when a
 * SAML assertion is made at runtime via the browser.
 */
```

On a related note, that comment is above the call to this method on ServiceDocumentManagerImpl

`public List<Dataverse> getDataversesUserHasPermissionOn(AuthenticatedUser user, Permission permission)`

Should this method be replaced by the new one in this PR:

`public List<Dataverse> findPermittedCollections(AuthenticatedUser user, int permissionBit)`

?

Reply:
I'll swap it out and test it

Reply:
@pdurbin I added IP Group support but unfortunately ServiceDocumentManagerImpl doesn't have access to the request and therefore the ip address of the caller. Unless you know of a way to get it?

Reply:
I don't see an obvious way. However, the SWORD API has never supported IP Groups and to my knowledge nobody has asked for it, so I think it's ok.

The main use case for IP Groups is read-only access, such as walking into a library and having access to data because you are on the library's IP range.